### DISCUSSION TITLE
Moment of Intertia in IfcOpenShell

### DISCUSSION THREAD

**Discussion by elisemunche** [IfcOpenShell/IfcOpenShell#3140] [Q&A]
Hello, 
I am writing my master thesis and need to extract the moment of intertia from an IFC file of IfcBeams and IfcColumns. Does anyone know how I can do this? 

Thank you in advance! 

-Elise Munch-Ellingsen

**Comment by aothms** [comment]
You can use `ifcopenshell.geom.create_shape(ifcopenshell.geom.settings(USE_PYTHON_OPENCASCADE=True), ...)` to get a python opencascade brep of a column or beam. And then use opencascade to calculate them https://dev.opencascade.org/doc/refman/html/class_g_prop___principal_props.html

It might be a bit of a puzzle but you'll figure it out. Once you get some results would be cool if you post back a summary of your findings either here or community.osarch.org/
